backend/src/db/utils.ts

```diff
@@ -1,40 +1,132 @@
 import { Knex } from "knex";
-
 import { TableName } from "./schemas";
 
-export const createJunctionTable = (knex: Knex, tableName: TableName, table1Name: TableName, table2Name: TableName) =>
-  knex.schema.createTable(tableName, (table) => {
-    table.uuid("id", { primaryKey: true }).defaultTo(knex.fn.uuid());
-    table.uuid(`${table1Name}Id`).unsigned().notNullable(); // Foreign key for table1
-    table.uuid(`${table2Name}Id`).unsigned().notNullable(); // Foreign key for table2
-    table.foreign(`${table1Name}Id`).references("id").inTable(table1Name).onDelete("CASCADE");
-    table.foreign(`${table2Name}Id`).references("id").inTable(table2Name).onDelete("CASCADE");
-  });
+interface JunctionTableOptions {
+  table1PrimaryKey?: string;
+  table2PrimaryKey?: string;
+  indexForeignKeys?: boolean;
+}
+
+export const createJunctionTable = async (
+  knex: Knex,
+  tableName: TableName,
+  table1Name: TableName,
+  table2Name: TableName,
+  options: JunctionTableOptions = {}
+): Promise<void> => {
+  const {
+    table1PrimaryKey = 'id',
+    table2PrimaryKey = 'id',
+    indexForeignKeys = false
+  } = options;
+
+  try {
```

Comment:
Please remove all try catch. The migration run in a transaction and if this fails the transaction should fail

Reply:
what if i don't remove try catch but throw the error, from the catch block

Reply:
It doesn't make sense to catch and then throw an error. The error from knex already contains proper context. Please remove it.

Reply:
removed all try and catch.